File: core/src/main/java/org/elkd/core/log/CommitCommand.java
package org.elkd.core.log;

import com.google.common.base.Preconditions;

import javax.annotation.Nonnull;

public class CommitCommand implements LogCommand<CommitResult<Entry>> {
  private final long mIndex;
  private final Log<Entry> mReceiver;
  private final LogChangeReason mReason;

  public CommitCommand(final long index,
                       @Nonnull final Log<Entry> receiver,
                       @Nonnull final LogChangeReason reason) {
    mIndex = index;
    mReceiver = Preconditions.checkNotNull(receiver, "receiver");
    mReason = Preconditions.checkNotNull(reason, "reason");
  }

  @Override
  public LogChangeReason getReason() {
    return mReason;
  }

  @Override
  public CommitResult<Entry> execute() {
    return mReceiver.commit(mIndex);
  }
}

File: core/src/main/java/org/elkd/core/log/commands/CommitCommand.kt
package org.elkd.core.log.commands

import org.elkd.core.log.LogChangeReason
import org.elkd.core.log.LogEntry
import org.elkd.core.log.ds.Log

class CommitCommand<E : LogEntry> constructor(
    private val index: Long,
    override val reason: LogChangeReason
) : LogCommand<E> {

  override fun execute(log: Log<E>) {
    log.commit(index)
  }

  companion object {
    @JvmStatic fun <E : LogEntry> build(index: Long, reason: LogChangeReason): CommitCommand<E> = CommitCommand(index, reason)
  }
}
